platform/posix/transport/src/openssl_posix.c

```diff
@@ -778,8 +790,28 @@ int32_t Openssl_Send( NetworkContext_t * pNetworkContext,
         {
             sslError = SSL_get_error( pOpensslParams->pSsl, bytesSent );
 
-            LogError( ( "Failed to send data over network: SSL_write of OpenSSL failed: "
-                        "ErrorStatus=%s.", ERR_reason_error_string( sslError ) ) );
+            if( sslError == SSL_ERROR_WANT_READ )
```

Comment:
[The documentation for SSL_write()](https://www.openssl.org/docs/man1.1.1/man3/SSL_write.html) says that it will only return `SSL_ERROR_WANT_READ` (and also `SSL_ERROR_WANT_WRITE`) in __nonblocking__ BIO implementations.  Also note that both of these return values are possible for `SSL_read()` as well with a nonblocking BIO.  So these comments as well as the ones in `Openssl_Recv()` don't seem consistent with the documentation.

Reply:
I had earlier checked against both `SSL_ERROR_WANT_READ` and `SSL_ERROR_WANT_WRITE` error codes, however, I was notified in [this comment](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1432#discussion_r537964636) that the transport implementation configured OpenSSL in blocking mode.

Reply:
[According to the documentation](https://www.openssl.org/docs/man1.1.1/man3/SSL_write.html), `SSL_Write()` will __not__ return either `SSL_ERROR_WANT_READ` or `SSL_ERROR_WANT_WRITE` in blocking mode, so lines 793-814 can be deleted since they will never execute.